Adds support for setting bottom padding

Tuoki opened the pull request:
I have not tested (and compile) this on iOS because I do not currently own any device which I could use for that purpose.

Tuoki committed:
Adds support for setting bottom padding

Zazo032 commented on lib/intercom_flutter.dart:
```diff
@@ -100,6 +100,10 @@ class Intercom {
     });
   }
 
```
Could you add a doc comment here explaining what does this method do? Something like:
```dart
/// Sets Intercom's chat `padding` (not really sure).
```

Tuoki replied:
Thank you for review. I implemented these wanted changes to new commit. I still cannot be sure that iOS implementation is correct because I have not tested (or compiled) these on iOS platform.

Zazo032 reviewed (commented):
Hi @Tuoki! Thanks for the contribution. Could you please add a test similar to the existing ones? Also, could you bump `pubspec.yaml`'s `version` to `2.2.1` and include your changes to `CHANGELOG.md`?

Tuoki committed:
Adds invokeMethod test for setting bottom padding

Also:
- Adds setBottomPadding method documentation
- Bumps version number from 2.2.0 to 2.2.1
- Updates CHANGELOG to contain latest changes

Zazo032 commented:
@v3rm0n @Spikes042 could you have a look to iOS code? Implementation seems trivial and ok to me, but I don't have enough experience to be sure about it. If valid, LGTM

Spikes042 commented on ios/Classes/IntercomFlutterPlugin.m:
```diff
@@ -69,6 +69,12 @@ - (void) handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
+    else if([@"setBottomPadding" isEqualToString:call.method]) {
+        NSInteger value = call.arguments[@"padding"];
+        CGFloat padding = (CGFloat)value;
+        [Intercom setBottomPadding:padding]
```
Missing terminating semicolon
```suggestion
        [Intercom setBottomPadding:padding];
```

Tuoki replied:
Good catch! I fixed that on new commit.

Tuoki committed:
Adds missing terminating semicolon

Zazo032 commented:
Thanks @Spikes042 @Tuoki! Will try to build the example on iOS today, and if it compiles, I'll merge it

Spikes042 commented:
The default launcher disappears when you call ```Intercom.setBottomPadding()``` on iOS. I have no idea why this would be happening but I tested only on iOS simulator which is infamously buggy. Android works as expected

Zazo032 commented:
You mean if you call `setBottomPadding` while chat screen is open?

Tuoki commented:
Thanks you for trying it! Some question / suggestions: 1. Which kind values did you try out? If on iOS side values were so big that launcher moved away from screen? 2. Setting bottom padding should also affect on position of in-app messages - did you try out to in-app messages? Do they come through even launcher disappear?

Spikes042 commented:
Nope. Just while the default launcher is visible.
Code to reproduce: 
```dart
// example/lib/main.dart

import 'package:flutter/material.dart';
import 'package:intercom_flutter/intercom_flutter.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Intercom.initialize(
    'appId',
    androidApiKey: 'androidApiKey',
    iosApiKey: 'iosApiKey',
  );
  Intercom.registerUnidentifiedUser();
  Intercom.setLauncherVisibility(IntercomVisibility.visible);
  runApp(SampleApp());
}

class SampleApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Intercom example app'),
        ),
        body: Center(
          child: FlatButton(
            onPressed: () {
              Intercom.setBottomPadding(50);
            },
            child: Text('Set padding'),
          ),
        ),
      ),
    );
  }
}

```

Once you tap Set Padding button, the launcher is moved to the new location (as expected) on Android but disappears on iOS and will never show up again even with hot restart until you terminate the app and build again.

Spikes042 commented:
@Tuoki 

1. I tried integer values 10, 50 and 100
2. iPhone 11 pro max has a resolution of 2436×1125 pixels so I doubt that. I tried on iPhone 8 as well. (Note, all on the simulator)
3. I'm not sure what I would be looking at in in-app messages but it does show up when you call ```Intercom.displayMessenger()```. I didn't see any difference even on Android

Tuoki commented:
It might be (i am just guessing here) that launcher is "visible" in sense that it is drawn to frame buffer but Flutter draws over it (as you know we cannot control Intercom SDK drawing) so if you try to send in-app message it should cause that Intercom redraws a notification box (message preview) and it should be on same level where laucher should be (+padding). That's why I asked you try out in-app messages. Actually this is the case where i need this in my current project because I draw custom launcher (which drawing I can control) and my issue is that I cannot set in-app message to correct position (because I cannot control Intercom drawing) without setting padding from bottom.

Tuoki commented:
@Zazo032 Is there possibility to go forward with this pull request or what we will do?

Zazo032 commented:
@Tuoki can't test this again right now - I won't have an available mac until mid-March approximately. Might this be an emulator-related issue, or a iOS sdk issue?

deepak786 commented:
@Zazo032 @Tuoki this is already implemented in https://github.com/v3rm0n/intercom_flutter/pull/114

Tuoki commented:
This feature has been already implemented in #114 so this can be closed
